### PR TITLE
fixes edge case in comparison check

### PR DESCRIPTION
**Pull request by fanatic75**
This PR fixes the edge case of comparing nested doc ID only if the nested level is less than MAX DEPTH allowed.

### CODE REVIEW

File: src/Database/Database.php [hunk 1]
```diff
@@ -2940,6 +2940,10 @@ public function updateDocument(string $collection, string $id, Document $documen
             foreach ($document as $key => $value) {
                 // Skip the nested documents as they will be checked later in recursions.
                 if (\array_key_exists($key, $relationships)) {
+                    // No need to compare nested documents more than max depth.
+                    if (count($this->relationshipWriteStack) >= Database::RELATION_MAX_DEPTH - 1) {
```
Comment (abnegate) [inline src/Database/Database.php:2944]: Let's keep this simple:

```suggestion
                    if (\count($this->relationshipWriteStack) > Database::RELATION_MAX_DEPTH) {
```

Comment (abnegate) [reply src/Database/Database.php:2944]: Ignore this for now, let's add a test for updating a 3rd level document and ensure the changes are made

File: tests/Database/Base.php [hunk 2]
```diff
@@ -3326,49 +3326,87 @@ public function testNoChangeUpdateDocumentWithRelationWithoutPermission(): void
...
-        $parent->setAttribute('children', ['child1']);
+        $updateLevel1 = static::getDatabase()->updateDocument('level1', 'level1', $level1);
 
-        $updatedParent = static::getDatabase()->updateDocument('parentRelationTest', 'parent1', $parent);
+        $this->assertEquals($level1, $updateLevel1);
 
-        $this->assertEquals($updatedParent->getUpdatedAt(), $parent->getUpdatedAt());
-        $this->assertEquals($updatedParent->getAttribute('children')[0]->getUpdatedAt(), $parent->getAttribute('children')[0]->getUpdatedAt());
+        $this->expectException(AuthorizationException::class);
+        static::getDatabase()->updateDocument('level1', 'level1', $level1->setAttribute('name', 'haha'));
```
Comment (abnegate) [inline tests/Database/Base.php:3405]: If you want to run code after this, you need to use a try/catch and instanceof on the exception, or fail the test if there was no exception thrown

File: tests/Database/Base.php [hunk 3]
```diff
@@ -3326,49 +3326,87 @@ public function testNoChangeUpdateDocumentWithRelationWithoutPermission(): void
...
 
-        static::getDatabase()->deleteCollection('parentRelationTest');
```
Comment (abnegate) [inline tests/Database/Base.php:3370]: Let's assert that we can successfully update a third level document

Comment (fanatic75) [reply tests/Database/Base.php:3370]: Added assertion

File: tests/Database/Base.php (hunk 2 above)
Comment (fanatic75) [reply tests/Database/Base.php]: fixed this

File: tests/Database/Base.php [hunk 4]
```diff
@@ -3326,49 +3326,93 @@ public function testNoChangeUpdateDocumentWithRelationWithoutPermission(): void
...
+        } catch(Exception $e) {
+            $this->assertInstanceOf(AuthorizationException::class, $e);
+        }
+        $level1->setAttribute('name', 'Level 1');
+        static::getDatabase()->updateCollection('level3', [
+            Permission::read(Role::any()),
+            Permission::create(Role::any()),
+            Permission::update(Role::any()),
+            Permission::delete(Role::any()),
+        ], false);
```
Comment (abnegate) [inline tests/Database/Base.php:3402]: You shouldn't be able to do this without update permission unless using `Auth::skip` 🤔 I'll address in a separate PR

**Review by abnegate — APPROVED** [review]

**Pull request merged**